Extension/src/LanguageServer/cppBuildTaskProvider.ts

```diff
@@ -178,7 +178,12 @@ export class CppBuildTaskProvider implements TaskProvider {
             const taskLabel: string = ((appendSourceToName && !compilerPathBase.startsWith(ext.configPrefix)) ?
                 ext.configPrefix : "") + compilerPathBase + " " + localize("build_active_file", "build active file");
             const programName: string = util.defaultExePath();
-            let args: string[] = isCl ? ['/Zi', '/EHsc', '/nologo', `/Fe${programName}`, '${file}'] : ['-fdiagnostics-color=always', '-g', '${file}', '-o', programName];
+            const isClang: boolean = !isCl && compilerPathBase.toLowerCase().includes("clang");
```

Comment:
This is not comprehensive.  On macOS, there are symbolic links from `gcc` that point to clang.  Perhaps we could indicate whether a compiler is gcc, clang or cl in the list of compilers returned from the native side, as the native side can determine which is clang and gcc based on defines returned when querying them.